garak/cli.py

```diff
@@ -498,20 +516,39 @@ def main(arguments=None) -> None:
             command.start_run()  # start the run now that all config validation is complete
             print(f"📜 reporting to {_config.transient.report_filename}")
 
-            if parsed_specs["detector"] == []:
+            if not _config.plugins.harnesses:
+                # Set the _config.plugins.harnesses
+                if parsed_specs["detector"] == []:
+                    _config.plugins.harnesses["Probewise"] = {}
+                else:
+                    _config.plugins.harnesses["Pxd"] = {}
+            
+            if "Probewise" in _config.plugins.harnesses:
                 command.probewise_run(
                     generator, parsed_specs["probe"], evaluator, parsed_specs["buff"]
                 )
-            else:
+            
+            if "Pxd" in _config.plugins.harnesses:
                 command.pxd_run(
                     generator,
                     parsed_specs["probe"],
                     parsed_specs["detector"],
                     evaluator,
-                    parsed_specs["buff"],
+                    parsed_specs["buff"]
                 )
 
             command.end_run()
+
+        elif "DetectorOnly" in _config.plugins.harnesses:
```

Comment:
Note that configuration format expects the module & class name:

```suggestion
        elif "detectoronly" in _config.plugins.harnesses:
```

``` json
{
  "detectoronly": 
  {
    "Detectoronly": 
    {
      "report_path": "<file_path>"
    }
  }
}
```